### PR TITLE
chore(integ): apply compareStackTemplate to the rest

### PR DESCRIPTION
**Pull request by iamhopaul123**
Preceding PR: https://github.com/aws/copilot-cli/pull/4110

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/deploy/cloudformation/stack/env_integration_test.go [hunk 1]
```diff
@@ -463,9 +463,13 @@ func compareStackTemplate(t *testing.T, actualObj, wantedObj map[any]any) {
 }
 
 func compareStackTemplateSection(t *testing.T, key, actual, wanted reflect.Value) {
-	require.True(t, actual.MapIndex(key).IsValid(),
+	actualExist, wantedExist := actual.MapIndex(key).IsValid(), wanted.MapIndex(key).IsValid()
```
Comment (paragbhingre) [inline internal/pkg/deploy/cloudformation/stack/env_integration_test.go:467]: ❤️

File: internal/pkg/deploy/cloudformation/stack/lb_grpc_web_service_integration_test.go [hunk 2]
```diff
@@ -103,7 +103,7 @@ func TestGrpcLoadBalancedWebService_Template(t *testing.T) {
 			expectedBytes := []byte(expected)
 			mExpected := make(map[interface{}]interface{})
 			require.NoError(t, yaml.Unmarshal(expectedBytes, mExpected))
-			require.Equal(t, mExpected, mActual)
+			compareStackTemplate(t, mExpected, mActual)
```
Comment (Lou1415926) [inline internal/pkg/deploy/cloudformation/stack/lb_grpc_web_service_integration_test.go:113]: Seems like `mActual` should go before `mExpected`
```suggestion
			compareStackTemplate(t, mActual, mExpected)
```

Comment (Lou1415926) [reply internal/pkg/deploy/cloudformation/stack/lb_grpc_web_service_integration_test.go:113]: I wonder why this is not complaining about `Metadata` being absent - `mExepcted` doesn't have `Metadata` it seems

Comment (iamhopaul123) [reply internal/pkg/deploy/cloudformation/stack/lb_grpc_web_service_integration_test.go:113]: https://github.com/aws/copilot-cli/pull/4155/files#diff-c34d543bad6d3e1ac6437fd95329f5694bd27e677804b1074f8371ec3c405933R447 i adjusted the order to be align with the golang test expect.

**Review by Lou1415926 — APPROVED** [review]

File: internal/pkg/deploy/cloudformation/stack/backend_svc_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+func TestBackendService_TemplateAndParamsGeneration(t *testing.T) {
```
Comment (dannyrandall) [inline internal/pkg/deploy/cloudformation/stack/backend_svc_integration_test.go:23]: should the existing test get deleted? in `backend_svc_test.go`

**Review by huanjani — APPROVED** [review]
LGTM!

File: internal/pkg/deploy/cloudformation/stack/env_integration_test.go [hunk 4]
```diff
@@ -463,9 +463,13 @@ func compareStackTemplate(t *testing.T, actualObj, wantedObj map[any]any) {
 }
 
 func compareStackTemplateSection(t *testing.T, key, actual, wanted reflect.Value) {
```
Comment (dannyrandall) [inline internal/pkg/deploy/cloudformation/stack/env_integration_test.go]: nit: these should prob be flipped as well

**Pull request merged**